src/Facebook/FacebookClient.php

```diff
@@ -203,32 +193,29 @@ public function sendRequest(FacebookRequest $request)
 
         list($url, $method, $headers, $body) = $this->prepareRequestMessage($request);
 
-        // Since file uploads can take a while, we need to give more time for uploads
-        $timeOut = static::DEFAULT_REQUEST_TIMEOUT;
-        if ($request->containsFileUploads()) {
-            $timeOut = static::DEFAULT_FILE_UPLOAD_REQUEST_TIMEOUT;
-        } elseif ($request->containsVideoUploads()) {
-            $timeOut = static::DEFAULT_VIDEO_UPLOAD_REQUEST_TIMEOUT;
-        }
-
-        // Should throw `FacebookSDKException` exception on HTTP client error.
-        // Don't catch to allow it to bubble up.
-        $rawResponse = $this->httpClientHandler->send($url, $method, $body, $headers, $timeOut);
+        $psr7Request = MessageFactoryDiscovery::find()->createRequest($method, $url, $headers, $body);
```

Comment:
Maybe my englesh is poor haha, I understood "Variable assignment redundant" like "assigned twice", but you mean "useless assignement"!
Isn't this a micro-optimization that will decrease clarity? :)